pkg/resource/apply.go

```diff
@@ -152,3 +208,20 @@ func (t *changeTracker) ToPrinter(name string) (printers.ResourcePrinter, error)
 		setResult:       f,
 	}, nil
 }
+
+func annotationTooLong(err error) bool {
+	if !errors.IsInvalid(err) {
+		return false
+	}
+	apiStatusErr, ok := err.(errors.APIStatus)
+	if !ok {
+		return false
+	}
+	causes := apiStatusErr.Status().Details.Causes
+	for _, c := range causes {
+		if c.Type == fieldTooLong && c.Field == "metadata.annotations" {
```

Comment:
Do we want to return true if and only if *all* of the errors are annotations too long? If there are other errors, then the create and update will presumably still fail.